src/Microsoft.AspNet.SignalR.JS/jquery.signalR.core.js

```diff
@@ -411,6 +411,11 @@
 
         clientProtocol: "2.0",
 
+        // We want to support older servers since the 2.0 change is to support redirection results, which isn't
+        // really breaking in the protocol. So if a user updates their client to 2.0 protocol version there's
+        // no reason they can't still connect to a 1.4 server.
+        supportedProtocols: [ "1.4", "2.0" ],
```

Comment:
isn't it 1.5 for js?
https://blog.3d-logic.com/2015/03/29/signalr-on-the-wire-an-informal-description-of-the-signalr-protocol/
> clientProtocol – the version of protocol used by the client. The most recent version is 1.5 however it is only used by the JavaScript client since the change that mandated bumping the version of the protocol to 1.5 is only relevant for this client. The .NET and C++ clients currently use version 1.4. Note that the server is designed to support down-level clients (i.e. clients using previous versions of the protocol) and the current (2.2.0) version supports protocol versions from 1.2 to 1.5

Reply:
And why we only support "1.4" only instead of "1.2 to 1.5" for the *1. New Client -> Old Server* scenario?

Reply:
The sever sends the client version back so 1.2 would never come back from this client.

Reply:
But we want to support the case when New Client(2.0) connects to Old Server(1.4 or others?), which can be 1.2?

Reply:
Not generally not, we don't support new clients connecting to old servers when there's a protocol change. Notice the client's minimum version isn't changing to 2.0 here. Newer clients can support both the redirect based negotiation urls (2.0) and 1.4 endpoints.

Reply:
Yep, this should be 1.5. Good catch, I'll update that.

As @davidfowl said, we normally do not permit a new client to connect to an old server (in the case of a protocol change). This is an exception because the protocol change is a very minor negotiation change and it's trivial to allow this in this case.